src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,40 @@
 
 class Animal {
   // write your code here
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+
+  bite(herbivor) {
+    if (herbivor instanceof Herbivore && herbivor.hidden === false) {
```

Comment:
```suggestion
    if (herbivor instanceof Herbivore && !herbivor.hidden) {
```
You could check if the hidden is false just by using this simplification of the statement, this would mean if herbivor.hidden is not true then proceed. :D basically the same but a little shortcut.